src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/OpenAPIBuilder.java

```diff
@@ -81,7 +82,7 @@ protected void addBasePathFormats(OpenAPI api) {
      * Returns the landing page document class. By default it returns {@link
      * AbstractLandingPageDocument} in case a service has more representations than usual
      */
-    protected Class getLandingPageDocumentClass() {
+    protected Class<AbstractLandingPageDocument> getLandingPageDocumentClass() {
```

Comment:
Seems like a subclass could return a subclass though. Shoud it be ``<? extends AbstractLandingPageDocument>``?

Reply:
Indeed, amended to return `<? extends AbstractLandingPageDocument>`